crates/ra_cargo_watch/src/lib.rs

```diff
@@ -352,8 +352,21 @@ impl WatchThread {
                 // It is okay to ignore the result, as it only errors if the process is already dead
                 let _ = child.kill();
 
-                // Again, we don't care about the exit status so just ignore the result
-                let _ = child.wait();
+                // Again, we are resilient to errors, so we don't try to panic here
+                match child.wait_with_output() {
+                    Ok(output) => match output.status.code() {
+                        Some(0) | None => {}
+                        Some(exit_code) => {
+                            let output =
+                                std::str::from_utf8(&output.stderr).unwrap_or("<bad utf8 output>");
+
+                            if !output.contains("could not compile") {
```

Comment:
let's rather check that exit code is equal to zero. Cargo uses `1` to mean "cli is wrong", and `101` for other errors  (like a wrong dependency). The are errors which we want to ignore, which are not "cound not compile".

Also, it seems like we should move waiting for the child into `run_cargo` function.

Reply:
@matklad 
If I move waiting for the child to `run_cargo()`, then we will send `CheckEvent::End` only after the child is released, or won't it be a problem?